src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs

```diff
@@ -128,12 +128,13 @@ public AnalyzerExecutor(AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateS
                         return DiagnosticAnalysisResultMap.Create(ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>.Empty, ImmutableDictionary<DiagnosticAnalyzer, AnalyzerTelemetryInfo>.Empty);
                     }
 
-                    session.AddAdditionalAssets(optionAsset);
+                    session.AddAdditionalAssets(optionAsset, cancellationToken);
```

Comment:
I saw the code below, yes, the InitializeAsync require cancellation for the session.